File: server/src/com/broll/gainea/server/core/objects/buffs/BuffableBoolean.java
package com.broll.gainea.server.core.objects.buffs;


public class BuffableBoolean<T> extends BuffableValue<T, BooleanBuff, Boolean> {

    public BuffableBoolean(T object, boolean value) {
        super(object);
        this.value = value;
        recalc();
    }

    protected void recalc() {
        buffedValue = value.booleanValue();
        for (BooleanBuff buff : buffs) {
            buffedValue = buff.getModifier();
        }
    }

}

File: server/src/main/com/broll/gainea/server/core/objects/buffs/BuffableBoolean.kt
package com.broll.gainea.server.core.objects.buffs

class BuffableBoolean<T>(target: T, value: Boolean) : BuffableValue<T, Boolean, BooleanBuff>(target, value) {
    init {
        this.value = value
        recalc()
    }

    override fun recalc() {
        buffedValue = rootValue
        for (buff in buffs) {
            buffedValue = buff.modifier
        }
    }
}
